File: struct-routes/src/Cmd.ts
export class Cmd {
    public route: string = null;
    public client: any = null;
    public server: any = null;

    public static create(route: string, client: any, server: any): Cmd {
        const ret: Cmd = new Cmd();
        ret.route = route;
        ret.client = client;
        ret.server = server;
        return ret;
    }

    public reg(recv: Map<string, Cmd>) {
        recv.set(this.route, this);
    }
}

File: unity-client/Assets/Plugins/Route/Cmd.cs
using System.Collections.Generic;
using System;

public class Cmd
{
    public string route = null;
    public Type client = null;
    public Type server = null;

    public Cmd(string route, Type client, Type server)
    {
        this.route = route;
        this.client = client;
        this.server = server;
    }

    public void Reg(Dictionary<string, Cmd> dict)
    {
        dict.Add(route, this);
    }
}